src/Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -1054,22 +1054,22 @@ private void AddComponentStats(StringBuilder statsBuilder)
                 statsBuilder.AppendLine($"Queueued downloads: {unassignedDownloads}");
             }
 
-            double averageBlockSizeKb = this.GetAverageBlockSizeBytes() / 1024.0;
+            double avgBlockSizeBytes = this.GetAverageBlockSizeBytes();
+            double averageBlockSizeKb = avgBlockSizeBytes / 1024.0;
             statsBuilder.AppendLine($"Average block size: {Math.Round(averageBlockSizeKb, 2)} KB");
 
-            double totalSpeedKB = (this.GetTotalSpeedOfAllPeersBytesPerSec() / 1024.0);
-            statsBuilder.AppendLine($"Total download speed: {Math.Round(totalSpeedKB, 2)} KB/sec");
-
-            // TODO: do that when component is activated.
-            // TODO move to nodestats and not bench
-            // just for logging
-            // show it as a part of nodestats, not separated spammer:
-            // avg download speed
-            // peer quality score (sort by quality score)
-            // number of assigned blocks
-            // MaxBlocksBeingDownloaded
-            // amount of blocks being downloaded
-            // show actual speed (no 1mb limit)
+            double totalSpeedBytesPerSec = this.GetTotalSpeedOfAllPeersBytesPerSec();
```

Comment:
Same here.

Reply:
It's ok here. In the console it will show NaN:

![untitled](https://user-images.githubusercontent.com/10259355/45818751-4e317280-bceb-11e8-8bba-5aca6eaba921.png)

Reply:
Sweet